cumulusci/tasks/bulkdata/tests/test_upsert.py

```diff
@@ -0,0 +1,400 @@
+import re
+from unittest import mock
+
+import pytest
+import responses
+
+from cumulusci.tasks.bulkdata import LoadData
+from cumulusci.tasks.bulkdata.step import DataApi, DataOperationStatus
+from cumulusci.tests.util import mock_describe_calls
+
+current_sf_version = "52.0"
+
+
+class TestUpsert:
+    # bulk API not supported by VCR yet
+    @pytest.mark.needs_org()
+    def test_upsert_external_id_field(
+        self, create_task, cumulusci_test_repo_root, sf, delete_data_from_org
+    ):
+        delete_data_from_org(["Entitlement", "Opportunity", "Contact", "Account"])
+        self._common_integration_testing_code(
+            "bulk", create_task, cumulusci_test_repo_root, sf
+        )
+
+    def _common_integration_testing_code(
+        self, api, create_task, cumulusci_test_repo_root, sf
+    ):
```

Comment:
Would be nice to have a brief indication of what scenario this method tests without needing to read it all. Simply renaming the method to `_run_load_data_task_and_check_for_upsert` would help.